src/Symfony/Component/SignalRegistry/SignalRegistry.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\SignalRegistry;
+
+final class SignalRegistry implements SignalRegistryInterface
+{
+    private $signals = [];
+
+    public function __construct()
+    {
+        pcntl_async_signals(true);
+    }
+
+    public function register(int $signal, callable $callback): void
+    {
+        if (!isset($this->signals[$signal])) {
+            $previousCallback = pcntl_signal_get_handler($signal);
```

Comment:
I'm not sure, but I think you should check if the previous callback is not this class.
Is it possible?

Reply:
No, they are all Closures at this place. 
When it may be helpful? I think it's fine to get all previous handlers.

Reply:
I just tested it, this is perfect :+1: